app/controllers/comments_controller.rb

```diff
@@ -8,43 +8,26 @@ def new
   # polycomment attributes (type, id) to help
   # tell what the comments are for
   def create
-    @comment = Comment.new comment_params
-    @comment.user = current_user
-    if @comment.save
-      @comments = Comment.where(
-        polycomment_type: params[:comment][:polycomment_type],
-        polycomment_id: params[:comment][:polycomment_id]
-      )
-      @comments = @comments.paginate(page: 1, per_page: 10)
-      project = Project.find_by(name: params[:comment][:project_name])
-
-      notify_followers(project, @comment)
-
-      respond_to do |format|
-        format.html { redirect_to :back }
-        format.js {}
+    if polycomment_exists
+      @comment = Comment.new comment_params
+      @comment.user = current_user
+      if @comment.save
+        @comments = find_project_comments
+        @project = Project.find_by(name: params[:comment][:project_name])
+        action = find_action
+        victims = @project.followers + [@project.user] - [@comment.user]
+        notify_users action, 1, @comment.id, victims
+        respond_to do |format|
+          format.html { redirect_to :back }
+          format.js {}
+        end
+      else
+        redirect_to :back
+        flash[:alert] = 'Something went wrong, try reposting your comment.'
       end
     else
-      redirect_to :back
-      flash[:alert] = 'Something went wrong, try reposting your comment.'
-    end
-  end
-
-  def notify_followers(project, comment)
-    if comment.polycomment_type == 'project'
-      action = 0
-    elsif comment.polycomment_type == 'issue'
-      action = 5
+      redirect_to root_url
```

Comment:
This is a bad request, I'm not comfortable with swallowing the error and sending a redirect. I think we should send a 4xx request code.. 404 seems apt.